### DISCUSSION TITLE
Error message: Particle [P1] from element [E1] lies outside the mesh

### DISCUSSION THREAD

**Discussion by JayfansChou** [Anura3D/Anura3D_OpenSource#247] [Q&A] [answered]
Dear all,

I'm Chou, and I'm learning MPM by using GiD 16.0.6 and Anura3D 2023. Recently, I've been using MPM method to calculate the bearing capacity coefficient of a deep-buried ball when it penetrates into homogeneous clay and compare it with theoretical solution and finite element results. The following is a detailed introduction to my model.

Model Dimension: 
a) unit (m, kPa), 2D axisymmetric, single_point
b) size: 30D*30D, where D = 0.113 m
![model dimension](https://github.com/user-attachments/assets/b2e9e0a3-23b5-45bd-b79a-2d92fbc1d8a4)

Material Properties: under undrained total stress analysis
a) soil: saturated clay, Mohr-Coulomb material
b) ball: metal, rigid body 
![material properties](https://github.com/user-attachments/assets/1db52a71-a2e4-4ef1-9f92-83f06ec38acf)

Material property distribution: divided into 4 areas
![material property assignment](https://github.com/user-attachments/assets/1bc6f416-f590-42cb-9822-c1dc6fc20b22)

Boundary conditions: only solid fixities
![boundary conditions](https://github.com/user-attachments/assets/41f90480-af8d-4f93-b908-e7aecc62757d)

Initial conditions: constant velocity 0.02m/s
![initial conditions](https://github.com/user-attachments/assets/6c29fd82-6213-4de4-800b-8a00f21119c6)

Contact properties: soil-ball interaction (rough contact)
![contact properties](https://github.com/user-attachments/assets/2e5aa135-2b3a-4378-a66a-10de53b27ff0)

Reaction Forces: the outside curve as contact faces
![reaction forces](https://github.com/user-attachments/assets/f5067be1-2e0d-4bf6-8c89-bc858dbc515a)

Moving mesh: 
a) extending mesh: upper soil
b) moving mesh: middle soil
c) compressing mesh: bottom soil
d) reference material: ball
![moving mesh](https://github.com/user-attachments/assets/1b65e69a-8493-45ff-91c5-73fb968ecdf1)

Mesh generation: uniform seed distribution
![mesh genetration](https://github.com/user-attachments/assets/b709e93d-250a-4955-8bb6-ad7799d7030c)

Calculate data: gravity load, K0 condition, etc
![calculate data1](https://github.com/user-attachments/assets/c112c0a7-1bd6-4da0-9005-d0ef1eaeddbc)
![calculate data2](https://github.com/user-attachments/assets/da0be516-5e00-43e7-88bf-54c2eba4628d)

Based on it, when I submit to calculation, it shows an error: [ERROR] : (time step: 99)Particle 164078 from element 54751 lies outside the mesh. X, Y= -0.5093E-002, 0.2744E+001. I'm wondering what's this reason and how should I solve it.
![error message](https://github.com/user-attachments/assets/344187a2-57dd-445c-b57c-b2bc7fe3bfcd)

I'd appreciate it for your help.

Best Regards, Chou

**Comment by Francisliii** [comment, marked as answer]
Hi,
Potential issues: 

1. Reduce courant number to less than 0.5 or smaller to reduce numerical instability.
2. You may need to apply node normal correction.
3. You have three layers, and in the k0-proceduce the number of layer should be 3.
4. Why would you put 0 as the density of liquid?
5. Give 5% damping.

You may need to use paraview to see where the exploding MP is.

Best,
Francis

**Reply by JayfansChou** [reply]
Dear Francis,

Thank you very much, I will have a try.

Best Regards,
Chou

**Comment by EmreDumlu-UM** [comment]
Reducing the courant number would definitely help. However, the error was occurred in the second timestep. I believe you should also check your fixities.

**Comment by DracuLaVai** [comment]
Dear JayfansChou,
Did you solve this problem? If yes, please share how it was solved. I am facing the same issue.
Thanks

**Comment by JayfansChou** [comment]
Dear DracuLaVai,




Unfortunately, I haven't been able to solve this problem before, and I haven't used the MPM method for some reasons. I'm sorry that I can't be of more help. Wishing you the best in your research.




Sincerely,

JayfansChou
















At 2025-03-30 01:52:04, "DracuLaVai" ***@***.***> wrote:

Dear JayfansChou,
Did you solve this problem? If yes, please share how it was solved. I am facing the same issue.
Thanks
